### PR TITLE
fix: resolve path compatibility issues on Windows devices

### LINKED ISSUE #877: `cdk synth` from a Windows host results in backslashes in paths.
When running `cdk synth` from my local machine, paths to the file assets in deploy.yml are generated with Windows-style backslashes:

`./cdk.out/assembly-OnstageCms\publish-Assets-FileAsset5-step.sh`

When I commit this file, it fails in my pipeline, because it detects that the deploy.yml is different from what it generated. I also assume these paths won't work even if the check somehow allowed this diff.

If I fix the paths manually, then commit and push, my pipelines deploy successfully.

Here's a sample from my deploy.yml with the issue:

```yaml
  Assets-FileAsset5:
    name: Publish Assets Assets-FileAsset5
    needs:
      - Build-Build-Cloudformation-Templates
    permissions:
      contents: read
      id-token: write
    runs-on: ubuntu-latest
    outputs:
      asset-hash: ${{ steps.Publish.outputs.asset-hash }}
    steps:
      - name: Download cdk.out
        uses: actions/download-artifact@v3
        with:
          name: cdk.out
          path: cdk.out
      - name: Install
        run: npm install --no-save cdk-assets
      - name: Authenticate Via OIDC Role
        uses: aws-actions/configure-aws-credentials@v4
        with:
          aws-region: us-west-2
          role-duration-seconds: 1800
          role-skip-session-tagging: true
          role-to-assume: arn:aws:iam::<redacted>:role/GithubActionsForStrapi
      - id: Publish
        name: Publish Assets-FileAsset5
        run: /bin/bash ./cdk.out/assembly-OnstageCms\publish-Assets-FileAsset5-step.sh
```

Environment: Windows 11
Node version: v18.18.2
Package manager: Yarn 4.1.0
cdk-pipelines-github: 0.4.110

### PR DESCRIPTION
**Pull request by EHitchc**
# tl;dr
Fixes #877

- Updates Jest config globs that did not match any tests on Windows device
- Resolves tests that fail when run on Windows platform
- Uses a helper `posixPath` function to convert all windows separators to posix instead of trying to find and replace all necessary instances of `path` module that would need to convert separators to be posix compliant

# More Info
Follow up to issue https://github.com/cdklabs/cdk-pipelines-github/issues/877 and original change https://github.com/cdklabs/cdk-pipelines-github/pull/880 which partially resolved issues

Additionally, fixed up Jest config so that tests run on Windows device:
Before updating `testMatch`:
```
$ yarn test
yarn run v1.22.21
$ npx projen test
👾 test | jest --passWithNoTests --coverageProvider=v8 --ci
No tests found, exiting with code 0
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |       0 |        0 |       0 |       0 |                  
----------|---------|----------|---------|---------|-------------------
```

After updating `testMatch`:
```
Snapshot Summary
 › 26 snapshots failed from 6 test suites. Inspect your code changes or re-run jest with `-u` to update them.

Test Suites: 7 failed, 2 passed, 9 total
Tests:       36 failed, 45 passed, 81 total
Snapshots:   26 failed, 13 passed, 39 total
Time:        39.922 s
Ran all test suites.
```

After posix path changes:
```
Test Suites: 9 passed, 9 total
Tests:       88 passed, 88 total
Snapshots:   43 passed, 43 total
Time:        40.989 s
Ran all test suites.
```

### CODE REVIEW

**Review by kaizencc — APPROVED** [review]
Thanks @EHitchc. Looks good and hopefully this resolves windows issues for now :)

**Comment by EHitchc** [comment]
Rebased due to resyncing fork with main, although it appears the build is being aborted due to the changes made to the test match regex

```
Error: Files were changed during build (see build log). If this was triggered from a fork, you will need to update your branch.
diff --git a/package.json b/package.json
index 827d3df..040[9](https://github.com/cdklabs/cdk-pipelines-github/actions/runs/8563142043/job/23506037649#step:8:10)69e 100644
--- a/package.json
+++ b/package.json
@@ -96,8 +96,7 @@
   "jest": {
     "testMatch": [
       "<rootDir>/src/**/__tests__/**/*.ts?(x)",
-      "<rootDir>/test/**/*(*.)@(spec|test).ts?(x)",
-      "<rootDir>/src/**/*(*.)@(spec|test).ts?(x)"
+      "<rootDir>/(test|src)/**/*(*.)@(spec|test).ts?(x)"
     ],
     "clearMocks": true,
     "collectCoverage": true,
Error: Process completed with exit code 1.
```

**Comment by EHitchc** [comment]
I could not seem to override the defaults completely, so I've instead appended the Windows compatible configs as additional testMatch patterns. Hopefully this resolves the build mutation issue

**Comment by kaizencc** [comment]
@EHitchc build was being aborted because of permission boundaries for first time contributors. not sure if your changes are relevant anymore
